### ISSUE TITLE
[Lint/EndAlignment] Align with start of line

### ISSUE THREAD

**Issue by rhys-vdw** [rubocop-hq/rubocop#4557]
I would prefer an option to align with the indentation of the first line containing the keyword, rather than special casing variable assignment (via the `variable` option).

```
Lint/EndAlignment:
  EnforcedStyleAlignWith: variable
```

I think this would probably cover most cases where `variable` is the chosen config also.

## Expected behavior

Permit the following:

```rb
[
  *case letters?
  when foo then bar
  else baz
  end
]
```

## Actual behavior

![image](https://user-images.githubusercontent.com/822105/27618362-5049a39c-5bff-11e7-9fae-bb9d89e5862b.png)

Wants this indentation:

```rb
[
  *case letters?
   when foo then bar
   else baz
   end
]
```

## RuboCop version

```
0.49.1 (using Parser 2.4.0.0, running on ruby 2.3.4 x86_64-darwin16)
```

**Comment by bbatsov** [comment]
Seems like a bug to me.

**Comment by jonas054** [comment]
There is a `start_of_line` option already:
https://github.com/bbatsov/rubocop/blob/v0.49.1/config/default.yml#L1391

I think that's what you want, @rhys-vdw. Removing the bug label.

**Comment by bbatsov** [comment]
> I think that's what you want, @rhys-vdw. Removing the bug label.

Still, I've never seen code starting with a `*` aligned the way RuboCop wants it to be aligned in this situation. It simply seems wrong to me...

**Comment by rhys-vdw** [comment]
Sorry guys, looks like I was looking at [old documentation](http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Lint/EndAlignment) that doesn't mention that option. Thanks for the help.

_Sent from my Samsung SM-A520F using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_
